### DISCUSSION TITLE
Cursor based ordering - using `orderBy` along with `after`

### DISCUSSION THREAD

**Discussion by jana19-dev** [prisma/prisma#2228] [Q&A] [answered]
I have a use case of fetching more items from the backend on page scroll. 
I can do that by using the `after` argument giving it the last item in my current state. 
But I am not getting the right results when combining it with `orderBy`. 

Is the `orderBy` happening at the end of query execution? I believe when combined with `after`, the `orderBy` should be done first and then apply the `after`. 

Example: 
```
query {
  getAllMyMedia (
    orderBy: {title: asc}
  ){
    id
    title
  }
}
```

```
[
  { "id": 152779, "title": null },
  { "id": 152780, "title": null },
  { "id": 152781, "title": null },
  ........
]
```

But when I provide `after`, I get empty results, but I expect to get the `152781` result. 
```
query {
  getAllMyMedia (
    orderBy: {title: asc}
    after: {id: 152780}
  ){
    id
    title
  }
}
```

```
[ ]
```

I am also not sure if this has something to do with some of the `title` fields being null in the database 🤔

**Comment by jana19-dev** [comment]
I am using `skip` (length of the current items) to fetch more instead of using `after` and it works as expected with `orderBy` 👍

**Comment by ryands17** [comment, marked as answer]
Hey @jana-uoft could you pls create a new bug report for this [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=)

**Reply by jana19-dev** [reply]
Sure, https://github.com/prisma/prisma/issues/2274

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
